lib/azure/armrest/armrest_manager.rb

```diff
@@ -167,6 +170,8 @@ def initialize(options = {})
 
         # Base URL used for REST calls. Modify within method calls as needed.
         @base_url = Azure::Armrest::RESOURCE
+
```

Comment:
I looked at the initialize code, seems `@client_id  = @@client_id || options.fetch(:client_id)` should favor `options` over `@@client_id`, same for the rest.